IdentityCore/src/util/NSDictionary+MSIDExtensions.m

```diff
@@ -27,64 +27,92 @@
 
 @implementation NSDictionary (MSIDExtensions)
 
++ (NSDictionary *)msidDictionaryFromString:(NSString *)string
+{
+    return [self msidDictionaryFromString:string decode:NO];
+}
+
 // Decodes a www-form-urlencoded string into a dictionary of key/value pairs.
 // Always returns a dictionary, even if the string is nil, empty or contains no pairs
 + (NSDictionary *)msidURLFormDecode:(NSString *)string
+{
+    return [self msidDictionaryFromString:string decode:YES];
+}
+
++ (NSDictionary *)msidDictionaryFromString:(NSString *)string
+                                    decode:(BOOL)decode
 {
     if (!string)
     {
         return nil;
     }
     
-    NSMutableDictionary *configuration = [[NSMutableDictionary alloc] init];
+    NSArray *queries = [string componentsSeparatedByString:@"&"];
+    NSMutableDictionary *queryDict = [NSMutableDictionary new];
     
-    if ( nil != string && string.length != 0 )
+    for (NSString *query in queries)
     {
-        NSArray *pairs = [string componentsSeparatedByString:@"&"];
+        NSArray *queryElements = [query componentsSeparatedByString:@"="];
+        if (queryElements.count > 2)
+        {
+            MSID_LOG_WARN(nil, @"Query parameter must be a form key=value");
```

Comment:
nit: also log the key of what got skipped if possible

Reply:
in an invalid query, we can't decide the key, may be we can print the query.